entity-service/src/main/kotlin/com/egm/stellio/entity/authorization/Neo4jAuthorizationRepository.kt

```diff
@@ -51,7 +54,13 @@ class Neo4jAuthorizationRepository(
             MATCH (userEntity:Entity { id: ${'$'}userId })
             OPTIONAL MATCH (userEntity)-[:HAS_VALUE]->(p:Property { name:"$EGM_ROLES" })
             OPTIONAL MATCH (userEntity)-[:HAS_OBJECT]-(r:Attribute:Relationship)-
-                [:IS_MEMBER_OF]->(group:Entity)-[:HAS_VALUE]->(pgroup:Property {name: "$EGM_ROLES"})
+                [:IS_MEMBER_OF]->(group:Entity)-[:HAS_VALUE]->(pgroup:Property { name: "$EGM_ROLES" })
+            RETURN p, pgroup
+            UNION
+            MATCH (client:Client)-[:HAS_VALUE]->(sid:Property { name: "$SERVICE_ACCOUNT_ID", value: ${'$'}userId })
```

Comment:
Since the value of the serviceAccountId property is stored under this format: urn:ngsi-ld:Client:serviceAccountId
I think that calling this method from the service with adding the USER_PREFIX to the SUB will not match any Client